server/SocketManager/socketmanager.js

```diff
@@ -1,11 +1,14 @@
 let sockets = {};
+let socket_user_map = {};
 
-const updateSocket = (usertoken, clientsocket) => {
+const updateSocket = (usertoken, clientsocket, user_id) => {
   sockets[usertoken] = clientsocket;
+  socket_user_map[usertoken] = user_id;
 };
 
 const removeSocket = usertoken => {
   delete sockets[usertoken];
+  delete socket_user_map[usertoken];
```

Comment:
In the `removeSocket` function, you are correctly deleting entries from both `sockets` and `socket_user_map`. However, it would be prudent to check if the `usertoken` exists in `socket_user_map` before attempting to delete it. This can prevent unnecessary operations and potential errors. You might want to add a check like this:
```javascript
if (usertoken in socket_user_map) {
    delete socket_user_map[usertoken];
}
```


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*